### PR TITLE
Safebrowsing v2

### LINKED ISSUE #646: safebrowsing upgrade to v2
@bbondy is there code that also needs to change? If I simply load V2 of this file (that is the only change I made):
https://s3.amazonaws.com/safe-browsing-data/2/SafeBrowsingData.dat
Then all sites I tried are coming back as positive hits (amazon, yahoo, etc).

### PR DESCRIPTION
**Pull request by garvankeeley**
fix #646 

Note the URL changed along with the version number

### CODE REVIEW

**Comment by garvankeeley** [comment]
hold on, adding a test case for this

File: brave/src/webfilters/NetworkDataFileLoader.swift [hunk 1]
```diff
@@ -89,8 +89,12 @@ class NetworkDataFileLoader {
             }
             else {
                 if let data = data, response = response as? NSHTTPURLResponse {
-                    let etag = response.allHeaderFields["Etag"] as? String
-                    self.finishWritingToDisk(data, etag: etag)
+                    if response.statusCode / 100 == 4 { // error
```
Comment (jhreis) [inline brave/src/webfilters/NetworkDataFileLoader.swift]: Any benefit of this over just checking `== 400`?

Comment (garvankeeley) [inline brave/src/webfilters/NetworkDataFileLoader.swift]: Any 4xx series error is bad

Comment (jhreis) [inline brave/src/webfilters/NetworkDataFileLoader.swift]: Ah, yes, can also use Swift's pattern matching `~=`:
`if 400...499 ~= response.statusCode { // error`

But I think either is fine.

Comment (garvankeeley) [inline brave/src/webfilters/NetworkDataFileLoader.swift]: that looks better, changing!

**Pull request merged**